src/test/java/seedu/address/model/book/BookNameTest.java

```diff
@@ -31,6 +31,12 @@ public void isValidBookName() {
         assertFalse(BookName.isValidBookName("Harry&Potter")); // contains non-alphanumeric characters not : or '
         assertFalse(BookName.isValidBookName(" HarryPotter")); // starts with non-alphanumeric character
 
+        // invalid book name but should be valid books
+        assertFalse(BookName.isValidBookName("Who Moved My Cheese?")); // ends with "?"
+        assertFalse(BookName.isValidBookName("4.50_from_Paddington")); // contains "."
+        assertFalse(BookName.isValidBookName("11/22/63")); // contains "/"
+        assertFalse(BookName.isValidBookName("Winnie-the-Pooh")); // contains "-"
```

Comment:
These may not be necessary because they are in the same equivalent partition as test case of line 31. It may incur a time cost of changing test cases if we change the regex to accept other characters in the future.